### PR TITLE
PLT-8212: Keep min length check while saving team name

### PR DESCRIPTION
**Pull request by pruthvip**
#### Summary
Minimum Length check on team name has been added when editing the name from Team settings

#### Ticket Link
[Github](https://github.com/mattermost/mattermost-server/issues/7874)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x ] Ran `make check-style` to check for style errors (required for all pull requests)
- [x ] Ran `make test` to ensure unit and component tests passed
- [x ] Has UI changes

### CODE REVIEW

**Comment by pruthvip** [comment]
Should I be using Utils.localizeMessage or FormattedMessage as used in the create team page is fine??

**Comment by pruthvip** [comment]
@esethna push the changes by evening

**Comment by pruthvip** [comment]
@esethna done

File: components/team_general_tab.jsx [hunk 1]
```diff
@@ -108,11 +108,25 @@ class GeneralTab extends React.Component {
...
-        } else {
+
+        if (!name) {
             state.clientError = Utils.localizeMessage('general_tab.required', 'This field is required');
             valid = false;
+        } else if (name.length < Constants.MIN_TEAMNAME_LENGTH) {
+            state.clientError = (
+                <FormattedMessage
+                    id='create_team.display_name.charLength'
+                    defaultMessage='Team Name must be {min} or more characters up to a maximum of {max}. You can add a longer team description later.'
```
Comment (jasonblais) [inline components/team_general_tab.jsx]: Sorry, can you also help add this string to en.json? https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json as per the [contribution guidelines](https://docs.mattermost.com/developer/contribution-guide.html#preparing-a-pull-request).

Thanks for helping with the fix!

Comment (pruthvip) [reply components/team_general_tab.jsx]: @jasonblais vl do it by evening

**Comment by jasonblais** [comment]
Hey @pruthvip, let us know if we can help with the above changes?

**Comment by pruthvip** [comment]
@jasonblais sorry for the delay. I have made the changes. Please have a look, if its fine.

**Comment by jasonblais** [comment]
Thanks @pruthvip! 

Not sure why, but the error text seems to be the old one, instead of the one proposed by @esethna: `Team Name must be 2 or more characters up to a maximum of 15. You can add a longer team description.`

image

**Comment by pruthvip** [comment]
@jasonblais hey thanks for pointing that out. I was making a mistake. I have fixed it now. But I still have an issue.  I am using `FormattedMessage` message to show the error, since `min` and `max` are dynamic values , I cant use `Utils.localizeMessage` directly. 

Hence in the render, `SettingItemMax` is expecting the error as a string, but  `FormattedMessage` returns an object. Which throws a prop mismatch error.  Can you guide me how to go about it.

**Comment by jasonblais** [comment]
@MusikPolice Can you help with this question?

> But I still have an issue.  I am using `FormattedMessage` message to show the error, since `min` and `max` are dynamic values , I cant use `Utils.localizeMessage` directly. 

> Hence in the render, `SettingItemMax` is expecting the error as a string, but  `FormattedMessage` returns an object. Which throws a prop mismatch error.  Can you guide me how to go about it. 

Functionally this looks good, thanks for the quick fixes!

File: components/team_general_tab.jsx [hunk 2]
```diff
@@ -108,11 +108,25 @@ class GeneralTab extends React.Component {
...
             valid = false;
+        } else if (name.length < Constants.MIN_TEAMNAME_LENGTH) {
+            state.clientError = (
+                <FormattedMessage
+                    id='general_tab.teamNameRestrictions'
+                    defaultMessage='Team Name must be {min} or more characters up to a maximum of {max}. You can add a longer team description.'
+                    values={{
+                        min: Constants.MIN_TEAMNAME_LENGTH,
+                        max: Constants.MAX_TEAMNAME_LENGTH
+                    }}
```
Comment (MusikPolice) [inline components/team_general_tab.jsx]: This looks like the right way to solve the problem. @jasonblais am I missing something, or is there still an issue outstanding here?

Comment (jasonblais) [reply components/team_general_tab.jsx]: @pruthvip asked a question about it. It looked good with testing.

Comment (jasonblais) [reply components/team_general_tab.jsx]: @MusikPolice We can merge if it looks good

**Comment by MusikPolice** [comment]
@jasonblais merged.

**Comment by pruthvip** [comment]
@MusikPolice The client_error which i pass to `SettingItemMax` is an object, because FormattedMessage gives back an object. But `SettingItemMax` is expecting a string as you can see in the propTypes.. Hence in development mode , react will throw an error of prop mismatch! Is this fine?

`SettingItemMax.propTypes = {
    inputs: PropTypes.array,
    client_error: PropTypes.string,
    server_error: PropTypes.string,
    extraInfo: PropTypes.element,
    infoPosition: PropTypes.string,
    updateSection: PropTypes.func,
    submit: PropTypes.func,
    saving: PropTypes.bool,
    title: PropTypes.node,
    width: PropTypes.string,
    submitExtra: PropTypes.node,
    cancelButtonText: PropTypes.node
};`

**Comment by MusikPolice** [comment]
@pruthvip it's certainly not ideal. I didn't realize that we still had this outstanding issue before I merged. Can you fix it in a separate PR? I suspect that you can pass some property of client error in to SettingItemMax - can you inspect it and see if it has a string field that contains the error message?

**Comment by pruthvip** [comment]
ya sure @MusikPolice . Vl look into that and vl raise a PR.

**Comment by pruthvip** [comment]
@MusikPolice , i have fixed the issue [https://github.com/mattermost/mattermost-webapp/pull/478](https://github.com/mattermost/mattermost-webapp/pull/478) . Have a look.

**Pull request merged**